contracts/external/cw-vesting/src/vesting.rs

```diff
@@ -420,6 +437,19 @@ impl Vest {
         self.status = Status::Canceled { owner_withdrawable };
         self.vested = Curve::Constant { y: self.vested(t) };
     }
+
+    /// Gets the duration of the vest. For constant curves, `None` is
+    /// returned.
+    pub fn duration(&self) -> Option<Duration> {
```

Comment:
if this will always be a `Duration::Time`, is there a reason to use a `Duration` type instead of a `u64` of the seconds?

on the frontend it adds extra steps (need to check `'time' in duration`) and an extra error handler to make sure it isn't `Duration::Blocks`

Reply:
🙏 bec15c70ad7095ab5fef9394618405275f32e2db

great feedback to hear. thanks.